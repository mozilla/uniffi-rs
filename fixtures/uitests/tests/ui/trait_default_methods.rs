#[uniffi::export]
pub trait Trait: Send + Sync {
    // a method with a default impl causes problems (#2597 etc)
    fn default(&self) -> String {
        unreachable!()
    }
}

fn main() { /* empty main required by `trybuild` */}
