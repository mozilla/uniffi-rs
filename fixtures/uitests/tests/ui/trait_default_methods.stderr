error: uniffi::export'd trait methods can't have a default implementation.
 --> tests/ui/trait_default_methods.rs:4:33
  |
4 |       fn default(&self) -> String {
  |  _________________________________^
5 | |         unreachable!()
6 | |     }
  | |_____^
